xdp/src/tx_loop.rs

```diff
@@ -199,19 +211,38 @@ pub fn tx_loop<T: AsRef<[u8]>>(
                     panic!("IPv6 not supported");
                 };
 
-                let next_hop = router.route(addr.ip()).unwrap();
-                // sanity check that the address is routable through our NIC
-                if next_hop.if_index != dev.if_index() {
-                    log::warn!(
-                        "turbine peer {} must be routed through if_index: {} our if_index: {}",
-                        addr,
-                        next_hop.if_index,
-                        dev.if_index()
-                    );
-                    batched_packets -= 1;
-                    umem.release(frame.offset());
-                    continue;
-                }
+                let dest_mac = if let Some(mac) = dest_mac {
+                    mac
+                } else {
+                    let next_hop = router.route(addr.ip()).unwrap();
+
+                    let mut skip = false;
+
+                    // sanity check that the address is routable through our NIC
+                    if next_hop.if_index != dev.if_index() {
+                        log::warn!(
+                            "turbine peer {addr} must be routed through if_index: {} our if_index: {}",
+                            next_hop.if_index,
+                            dev.if_index()
+                        );
+                        skip = true;
+                    }
+
+                    // we need the MAC address to send the packet
+                    if next_hop.mac_addr.is_none() {
```

Comment:
previously, if sending to loopback ifc, we would just populate a dummy mac addr, right? Now we will warn and drop the packet.

Seems fine to me, but wanted to double check

Reply:
yes, the reasoning is the following:

the validator never wants to send to loopback; for testing loopback (which I sometimes do), I can now pass an explicit `dest_mac` to the function